apex/transformer/testing/commons.py

```diff
@@ -94,6 +202,36 @@ def loss_func(x):
     return y, loss_func
 
 
+@dataclass(frozen=True)
+class ToyParallelMLPFwdBwdStepFunc:
+
+    sequence_parallel_enabled: bool
+
+    def __call__(
+        self,
+        batch: Batch,
+        model: torch.nn.Module,
+    ) -> Tuple[torch.Tensor, Callable[[torch.Tensor], Tuple[torch.Tensor, Dict[str, torch.Tensor]]]]:
+        x = batch[0] if isinstance(batch, list) else batch
+        if isinstance(x, torch.Tensor):
+            x = x.transpose(0, 1).contiguous()
+            if self.sequence_parallel_enabled:
+                x = x.chunk(
+                    chunks=transformer.parallel_state.get_tensor_model_parallel_world_size(),
+                    dim=0
+                )[transformer.parallel_state.get_tensor_model_parallel_rank()].contiguous()
```

Comment:
is this more concise?
```suggestion
                x = tensor_parallel.mappings.scatter_to_sequence_parallel_region(x)
```